### ISSUE TITLE
backupccl: backup with revision history does not work during un-finalized upgrade from 20.2 to 21.1.18

### ISSUE THREAD

**Issue by stevendanna** [cockroachdb/cockroach#73043]
`BACKUP ... WITH revision_history` fails with the following error when initiated from a cluster that has been upgrade to the 21.1.18 binary but has not yet had the cluster version finalized:

```
BACKUP INTO 'nodelocal://3//backup4' WITH revision_history;
ERROR: descriptor 40 not found: descriptor not found
```

This error is returned while we try to build a table of system tables that ought be opted out of the backup:

https://github.com/cockroachdb/cockroach/blob/d785088de79cdec6db17fd53067d204188d23973/pkg/ccl/backupccl/system_schema.go#L374-L390

This code assumes that when it goes to look up a table that not exists, it would receive a nil error and `found == false`.  But `GetMutableTableByName` returns a non-nil error in at least some of the cases where the descriptor is not found.

**Comment by stevendanna** [comment]
Not this also applies to the upgrades from 21.1 to the (unrelease) 21.2.1+

**Comment by adityamaru** [comment]
This has been fixed in #73110.
